src/geom/prep/PreparedPolygonPredicate.cpp

```diff
@@ -37,71 +38,118 @@ namespace prep { // geos.geom.prep
 //
 // protected:
 //
-bool
-PreparedPolygonPredicate::isAllTestComponentsInTarget(const geom::Geometry* testGeom) const
-{
-    geom::Coordinate::ConstVect pts;
-    geom::util::ComponentCoordinateExtracter::getCoordinates(*testGeom, pts);
-
-    for(std::size_t i = 0, ni = pts.size(); i < ni; i++) {
-        const geom::Coordinate* pt = pts[i];
-        const int loc = prepPoly->getPointLocator()->locate(pt);
-        if(geom::Location::EXTERIOR == loc) {
-            return false;
+struct AnyMatchingLocationFilter : public GeometryComponentFilter {
```

Comment:
How about calling these `LocationNotMatchingFilter` and `LocationMatchingFilter`?

Shorter, and sorts better.

Reply:
Could also combine into one filter, with a flag for the sense of the matching test.

Reply:
Done in eb34085